services/app/src/components/views/QuestHistory.tsx

```diff
@@ -0,0 +1,93 @@
+import * as React from 'react';
+import {QuestDetails} from '../../reducers/QuestTypes';
+import {SelectionListPhase, UserQuestInstance, UserQuestsType} from '../../reducers/StateTypes';
+import Button from '../base/Button';
+import Card from '../base/Card';
+
+const Moment = require('moment');
+
+export interface QuestHistoryStateProps {
+  phase: SelectionListPhase;
+  played: UserQuestsType;
+  selected: UserQuestInstance|null;
+}
+
+export interface QuestHistoryDispatchProps {
+  onSelect: (selected: UserQuestInstance) => void;
+  onPlay: (details: QuestDetails) => void;
+  onReturn: () => any;
+}
+
+export interface QuestHistoryProps extends QuestHistoryStateProps, QuestHistoryDispatchProps {}
+
+function renderDetails(props: QuestHistoryProps): JSX.Element {
```

Comment:
Is this identical to the search renderDetails function? If we're making another duplicate, can we instead de-duplicate it and close https://github.com/ExpeditionRPG/expedition/issues/258 ?

Reply:
Nope, there's several areas that are slightly different. Might make sense for this to become a proper react component eventually, though.